library/asn1parse.c

```diff
@@ -164,6 +180,11 @@ int mbedtls_asn1_get_int( unsigned char **p,
         (*p)++;
     }
 
+#if defined(MBEDTLS_ASN1_PARSE_NEGINT)
+    if (neg)
+        *val = ( ~(0) << ((*p - neg) * 8 - 1) ) | *val;
```

Comment:
I find this hard to read. Suggestion: The arithmetic `*p - neg`, with `neg` being the original value of `*p`, is unnecessary - instead, `neg` could be declared as `int` and set to `-1 << ( ( 8 * len ) - 1 )` if the sign-bit is set, and to `0` otherwise. Then, a uniform `*val += neg` can accomodate for the sign-bit in the end. Compilers will be clever enough to optimize away `neg` in case `MBEDTLS_ASN1_PARSE_NEGINT` is not set, and therefore `neg = 0` statically.